src/mempool.rs

```diff
@@ -203,6 +207,20 @@ impl Mempool {
                     }
                 }
             }
+            listen<DataEvent> cmd => {
+                 if let DataEvent::NewBlock(block) = cmd {
+                    for contract in block.new_contract_txs {
+                        let TransactionData::RegisterContract(register_contract_transaction) = contract.transaction_data else {
+                            continue;
+                        };
```

Comment:
Remarque: dans ce cas particulier je pense le `if let` fait bien le café mais no need to change

Reply:
oui en général je suis pas contre mais je sais pas si on gagne vraiment en lisibilité là :P